addons/website_hr_recruitment/tests/test_website_hr_recruitment.py

```diff
@@ -7,17 +7,29 @@
 @odoo.tests.tagged('post_install', '-at_install')
 class TestWebsiteHrRecruitmentForm(odoo.tests.HttpCase):
     def test_tour(self):
-        job = self.env['hr.job'].create({
-            'name': 'A Test Job',
+        job_guru = self.env['hr.job'].create({
+            'name': 'Guru',
             'is_published': True,
         })
-
+        job_intern = self.env['hr.job'].create({
+            'name': 'Internship',
+            'is_published': True,
+        })
+        self.start_tour('/', 'website_hr_recruitment_tour_edit_form', login='admin')
         with odoo.tests.RecordCapturer(self.env['hr.applicant'], []) as capt:
-            self.start_tour("/", 'website_hr_recruitment_tour')
+            self.start_tour('/', 'website_hr_recruitment_tour')
 
         # check result
-        record = capt.records
-        self.assertEqual(len(record), 1)
-        self.assertEqual(record.partner_name, "John Smith")
-        self.assertEqual(record.email_from, "john@smith.com")
-        self.assertEqual(record.partner_phone, '118.218')
+        guru_applicant = capt.records[1]
+        self.assertEqual(len(guru_applicant), 1)
```

Comment:
Do you know what this `RecordCapturer` change is all about? Maybe speeding up test? Seems to make it less readable and prone to mistake, like the one you do here: you get *one* record then verify that there is only one? If the fix this tours is enforcing is removed, don't you get 2 applications for the same job and thus here you will say "1 = 1 ok" instead of "2 != 1 fail" ?

Reply:
Indeed, this line doesn't make sense anymore, my bad. The RecordCapturer just seems to be a helper, [the commit that introduced it](https://github.com/odoo/odoo/commit/9a8cf18ddbfb11ac665b31cbefc2340a505ed3c9) doesn't talk about any performance gain and looking quickly at the code of this one it doesn't seem to bring any advantage for this test... so we can probably not use it and keep the test as it was.

Reply:
@Julien00859 Maybe you can enlighten us here? :+1:

Reply:
The RecordCapturer is about capturing the records that have been created for a specific model while in the content manager. Here you run a tour that'll create a record, you want to verify that that record have been created with the correct datas.

Without the RecordCapturer when doing a single search, there is a chance a record whose description is `'### [GURU] HR RECRUITMENT TEST DATA ###'` was existing in the database before running the test. I 100% agree you crafted this description so that is unique and searchable. Still, using RecordCapturer we don't have to craft such unique strings to make the test robust.

While I was busy doing httpocalypse I introduced a bug that added "<p>" and "</p>" around the description thus no record was found using the domain `('description', '=', '### [GURU] HR RECRUITMENT TEST DATA ###')`. It took me some time to understand what was going on, I decided to change the test so it uses RecordCapturer so that the assertion message was "wrong description" instead of "missing record".

Reply:
Thanks for the explanation, I agree with the idea behind `RecordCapturer`. I guess we can use it @Guillaume-gdi : instead of the two searches with one assertion for each retrieved records, capture the records during the tour (as it currently is now) and do only one assertion checking there are two records. Then for each one, ensure it is the correct job_id.

> [...] I decided to change the test so it uses RecordCapturer so that the assertion message was "wrong description" instead of "missing record".

I don't see that "wrong description" assertion in your updated code @Julien00859 so I am not sure to understand that part :relaxed: @Guillaume-gdi you add add an extra assert by record to ensure the description is right too.

![image](https://user-images.githubusercontent.com/10338094/165039040-a071e3b4-12fa-437f-8f1f-538f0e624e5a.png)

Reply:
> While I was busy doing httpocalypse I introduced a bug that added `<p>` and `</p>` around the description thus no record was found using the domain `('description', '=', '### [GURU] HR RECRUITMENT TEST DATA ###')`. It took me some time to understand what was going on, I decided to change the test so it uses RecordCapturer so that the assertion message was "wrong description" instead of "missing record".

It seems that the bug you are talking about still exists
When a user fills out the job application form, what he puts in the textarea description ends up in a `<p>` in the db and so here when we do capt.records[0].description we have the tag `<p>`.
Are you sure this is a bug and we shouldn't just do the check with the `<p>` ?
![image](https://user-images.githubusercontent.com/78849981/165084971-579cdcd5-0be7-443f-9969-5660084709a3.png)

see: https://runbot.odoo.com/runbot/build/14851757

Reply:
@Julien00859 More info about this? Not sure how your httpocalypse branch could have started wrapping stuff in `<p/>` tbh. @Guillaume-gdi did you investigate a bit? (was it always like that? is not it normal since the use of the editor and the fact that the field is an HTML one? ... :)

Reply:
> @Julien00859 More info about this? Not sure how your httpocalypse branch could have started wrapping stuff in `<p/>` tbh. @Guillaume-gdi did you investigate a bit? (was it always like that? is not it normal since the use of the editor and the fact that the field is an HTML one? ... :)

All I know for now is that in 14.0 there was no wrapping in `<p>` and from 15.0 on there is wrapping... I'll look into it a bit more

Reply:
Indeed in saas 14.4 there was [this commit](https://github.com/odoo/odoo/commit/b8d5b61333b0e8570ea2eb0535e37a0025514812) that transforms the text coming from the textarea into HTML.
What is strange IMHO is when in DB we have a hr.applicant with a description = `<p>### [GURU] HR RECRUITMENT TEST DATA ###</p>` and we make
`self.env['hr.applicant'].search([('description', '=', '### [GURU] HR RECRUITMENT TEST DATA ###')])` we find the element...
on the other hand if we do 
```python
applicant = self.env['hr.applicant'].search([('description', '=', '### [GURU] HR RECRUITMENT TEST DATA ###')])
print(applicant['description'])
```
It prints what is in the DB => `<p>### [GURU] HR RECRUITMENT TEST DATA ###</p>`, this new RecordCapturer returns the record like the search also with the `<p>`.

Reply:
This may seems very strange at first, but there is a reason why
![image](https://user-images.githubusercontent.com/30048408/165178603-590b7913-880a-4fcc-ac88-724f099f66b6.png)
is actually the same.

If you dig through the ORM code with a debugger, you will see that when doing 
```py
self.env['hr.applicant'].search([('description', '=', '### [GURU] HR RECRUITMENT TEST DATA ###')])
```
The ORM actually ends up calling `html_sanitize()` which wrap that description in a `<p/>` before sending that where condition to SQL.
![image](https://user-images.githubusercontent.com/30048408/165179740-102f25db-8616-4dd3-8c76-a066302b5484.png)

Here is the call stack
```
def search()
  def _search()
    def _where_calc()
      expression.expression()
      > expression.__init__()
          self.parse()
            self.__leaf_to_sql()
              params = [field.convert_to_column(right, model, validate=False)]
                Class Html.convert_to_column()
                  return html_sanitize()
                    def html_sanitize()
```

And this is because the field is not declared with `sanitize=False` (or something like that, you don't have to dig through everything, just figure the global picture). In this case, you just check with another field like `blog.post.content` and you see that you don't observe the same behavior.

For instance, a quick check if you create those 2 fields:
![image](https://user-images.githubusercontent.com/30048408/165179973-97bfe61f-9332-42fa-a80b-744589d9808d.png)
You end up with
![image](https://user-images.githubusercontent.com/30048408/165180069-31a9e78e-024d-459d-ab89-4229dd2686c7.png)

From there, without understanding everything you can just assume that everything is probably working as expected.
(If you want to ensure that or prove differently, you have to spend more than 15 minutes as I just did to give you an idea of how you could have debug that part).